identity/src/main/java/com/stripe/android/identity/navigation/ErrorFragment.kt

```diff
@@ -41,38 +44,40 @@ internal class ErrorFragment(
-                SCREEN_NAME_ERROR
-            )
-            bottomButton.setOnClickListener {
-                verificationFlowFinishable.finishWithResult(
-                    Failed(cause)
-                )
-            }
-        } else {
-            bottomButton.setOnClickListener {
-                identityViewModel.screenTracker.screenTransitionStart(
-                    SCREEN_NAME_ERROR
-                )
-                val destination = args[ARG_GO_BACK_BUTTON_DESTINATION] as Int
-                if (destination == UNEXPECTED_DESTINATION) {
-                    findNavController().navigate(DEFAULT_BACK_BUTTON_NAVIGATION)
-                } else {
-                    findNavController().let { navController ->
-                        var shouldContinueNavigateUp = true
-                        while (shouldContinueNavigateUp && navController.currentDestination?.id != destination) {
-                            shouldContinueNavigateUp =
-                                navController.navigateUpAndSetArgForUploadFragment()
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            ErrorScreen(
+                title = requireNotNull(args.getString(ARG_ERROR_TITLE)),
+                message1 = requireNotNull(args.getString(ARG_ERROR_CONTENT)),
+                bottomButton = (
+                    requireNotNull(args.getString(ARG_GO_BACK_BUTTON_TEXT)) to {
+                        identityViewModel.screenTracker.screenTransitionStart(
+                            SCREEN_NAME_ERROR
+                        )
+                        if (args.getBoolean(ARG_SHOULD_FAIL, false)) {
+                            verificationFlowFinishable.finishWithResult(
+                                IdentityVerificationSheet.VerificationFlowResult.Failed(cause)
+                            )
+                        } else {
+                            val destination = args.getInt(ARG_GO_BACK_BUTTON_DESTINATION)
+                            if (destination == UNEXPECTED_DESTINATION) {
+                                findNavController().navigate(DEFAULT_BACK_BUTTON_NAVIGATION)
+                            } else {
+                                findNavController().let { navController ->
+                                    var shouldContinueNavigateUp = true
+                                    while (
+                                        shouldContinueNavigateUp &&
+                                        navController.currentDestination?.id != destination
+                                    ) {
+                                        shouldContinueNavigateUp =
+                                            navController.navigateUpAndSetArgForUploadFragment()
+                                    }
+                                }
+                            }
                         }
                     }
-                }
-            }
+                    )
```

Comment:
Nit: Some indentation issues. Do we run ktlint or detekt in the Identity module?

Reply:
hmm actually detekt formatted it this way

Reply:
with the new `ErrorScreenButton` refactor this seems normal now